### PR TITLE
Use fixed keys for supplementary data list mappings

### PR DESCRIPTION
**Pull request by katie-gardner**
### What is the context of this PR?
This PR supports the use of integer identifiers for supplementary list items. Previously this was an issue as we validate the identifiers are strings, and whilst python supports integer keys in a dictionary, JSON serialisation will convert the keys to strings. This updates the validation to allow either, and adjusts the output of `supplementary_list_mappings` to be a list of objects with `identifier` and `list_item_id` keys with corresponding values.

### How to review
Check that the data being sent downstream matches the definition in [the documentation PR](https://github.com/ONSdigital/ons-schema-definitions/pull/86). You can try running the mock sds locally and changing identifiers to ints and checking that all still works as expected. The dump endpoint should show the list mappings in the new correct format

### Checklist
* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by katie-gardner** [comment]
[Prevent pre-populated lists from being modified (2 SP)](https://trello.com/c/w8LhMHet/5785-prevent-pre-populated-lists-from-being-modified-2-sp)

**Comment by katie-gardner** [comment]
[Update downstream supplementary list mappings to use non arbitrary keys](https://trello.com/c/Jqwvhit3/5844-update-downstream-supplementary-list-mappings-to-use-non-arbitrary-keys)

**Review by MebinAbraham — COMMENTED** [review]
Functionally all good, just some minor comments.

File: app/data_models/questionnaire_store.py [hunk 1]
```diff
@@ -87,23 +90,27 @@ def set_supplementary_data(self, to_set: MutableMapping) -> None:
...
     def _create_supplementary_list(
         self, *, list_name: str, list_data: list[dict]
-    ) -> dict[str, str]:
+    ) -> list[SupplementaryDataListMapping]:
         """
         Creates or updates a list in ListStore based off supplementary data
         returns the identifier -> list_item_id mappings used
         """
-        list_mapping: dict[str, str] = {}
+        list_mapping: list[SupplementaryDataListMapping] = []
```
[the remark this answers is not in the document]
Comment (katie-gardner) [reply app/data_models/questionnaire_store.py]: Yep, missed that one, thanks 👍

File: app/data_models/supplementary_data_store.py [hunk 2]
```diff
@@ -23,14 +28,27 @@ class SupplementaryDataStore:
...
         """
         self._raw_data = supplementary_data or {}
         self._list_mappings = list_mappings or {}
+        # create a lookup for easily finding the list_item_id for a given identifier
+        self._list_lookup = {
+            list_name: {
+                mapping["identifier"]: mapping["list_item_id"] for mapping in list_data
+            }
+            for list_name, list_data in self._list_mappings.items()
+        }
```
[the remark this answers is not in the document]
Comment (katie-gardner) [reply app/data_models/supplementary_data_store.py]: Yeah good idea, done

**Review by MebinAbraham — APPROVED** [review]
👍

**Review by Yuyuutsu — COMMENTED** [review]
Changes looks good just currently testing the functionality 👍

**Review by Yuyuutsu — APPROVED** [review]

**Pull request merged**